### ISSUE TITLE
Stability

### ISSUE THREAD

**Issue by danielfaust** [googlesamples/identity-toolkit-android#10]
How stable is this gitkit android API?

I've got the web app version working, but the Android programming experience is awful. The documentation is inconsistent, and even the demo doesn't work properly. Not to mention that the facebook plugin is using a deprecated API (3.xxx) instead of (4.4.0).

Is it normal that the demo app takes 1 minute and 5 seconds to log in after a fresh start, and then usually 5 seconds, but sometimes the 15 seconds and up to 1 minute and 5 seconds again?

Where's the source code to see what's going on?

**Comment by jin-liu** [comment]
Any latency longer than 5 seconds is definitely not the expected performance. Do you see any pattern for the slow case, like
- Using a real device or emulator
- Android OS version
- Time
- Location
- Other?

**Comment by danielfaust** [comment]
Hi jin-liu, thanks for the answer.

I found the culprit: The intranet I'm in is adding IPv6 support, but any outgoing traffic is still blocked at the edge of the network.

The DNS server is handing out AAAA responses, so apparently at the first start of the app it attempts to connect via IPv6, times out after a minute, then retries with IPv4 and succeeds after 5 seconds (at max). This happens always on the first start of the app, after "killing" it, completely reproducible (and sometimes even while it's running, that is at random).

I had been noting an odd behaviour for some days now, including non-loading images in other apps.

I installed "[ROOT] Disable IPv6" https://play.google.com/store/apps/details?id=de.lennartschoch.disableipv6 and after disabling IPv6 it worked immediately.

While I think that this is an Android issue, maybe GITkit should attempt IPv4 first, and then IPv6 if IPv4 fails or maybe there's another method to detect a non-functional IPv6 network.

I think that this is generally a serious issue, since many affected by such a problem, maybe through a misconfigured network, will most probably not be able to find the problem and solve it.

Is the source code avaliable somewhere?

Kind regards,
Daniel
